packages/block-editor/src/components/block-tools/index.js

```diff
@@ -20,6 +20,17 @@ import { store as blockEditorStore } from '../../store';
 import BlockContextualToolbar from './block-contextual-toolbar';
 import { usePopoverScroll } from './use-popover-scroll';
 
+/**
+ * Returns true if the container contains the document active element.
+ *
+ * @param {HTMLElement} container An HTML element.
+ *
+ * @return {boolean} Whether the container contains the currently active element in the document.
+ */
+const hasFocusWithin = ( container ) => {
+	return !! container?.contains( container?.ownerDocument?.activeElement );
```

Comment:
Or maybe we should be defensive here instead?
```suggestion
	return !! container && !! container?.contains( container?.ownerDocument?.activeElement );
```

Reply:
I quite like the defensive conditional you added in `onKeyDown`, so I think it'd be fine to include the defensive in both places?